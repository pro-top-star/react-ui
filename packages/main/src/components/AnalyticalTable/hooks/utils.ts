export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
