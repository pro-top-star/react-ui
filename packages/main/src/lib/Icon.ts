import { Icon } from '../webComponents/Icon';
import type { IconPropTypes } from '../webComponents/Icon';

export { Icon };
export type { IconPropTypes };
