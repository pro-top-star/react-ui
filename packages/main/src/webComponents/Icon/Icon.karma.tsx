import { mountThemedComponent } from '@shared/tests/utils';
import { expect, use } from 'chai';
import { matchSnapshot } from 'chai-karma-snapshot';
import React from 'react';
import { Icon } from '../../lib/Icon';

use(matchSnapshot);

describe('Icon', () => {
  it('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<Icon />);
    expect(wrapper.debug()).to.matchSnapshot();
  });
});
