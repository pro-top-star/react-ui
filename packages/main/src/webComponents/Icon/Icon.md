[Link to UI5 Web Components Playground](https://sap.github.io/ui5-webcomponents/playground/components/Icon)

```jsx
import { Icon } from '@ui5/webcomponents-react/lib/Icon';
```
