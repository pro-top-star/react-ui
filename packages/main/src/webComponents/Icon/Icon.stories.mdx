import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs/blocks';
import { Icon } from '@ui5/webcomponents-react/lib/Icon';
import '@ui5/webcomponents-icons/dist/employee.js';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { DocsHeader } from '@shared/stories/DocsHeader';
import { DocsCommonProps } from '@shared/stories/DocsCommonProps';

<Meta
  title="UI5 Web Components / Icon"
  component={Icon}
  argTypes={{
    ...createSelectArgTypes({}),
    ...DocsCommonProps
  }}
  args={{
    name: 'employee',
    style: {},
    className: '',
    tooltip: '',
    slot: '',
    ref: null
  }}
/>

<DocsHeader />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      return <Icon {...args} />;
    }}
  </Story>
</Canvas>

## Properties

<ArgsTable story="." />
