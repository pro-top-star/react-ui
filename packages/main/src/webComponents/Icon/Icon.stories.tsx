import { boolean, text } from '@storybook/addon-knobs';
import '@ui5/webcomponents-icons/dist/icons/add';
import { Icon } from '@ui5/webcomponents-react/lib/Icon';
import React from 'react';

export default {
  title: 'UI5 Web Components / Icon',
  component: Icon
};

export const generatedDefaultStory = () => (
  <Icon
    accessibleName={text('accessibleName', '')}
    name={text('name', 'add')}
    showTooltip={boolean('showTooltip', false)}
  />
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};
