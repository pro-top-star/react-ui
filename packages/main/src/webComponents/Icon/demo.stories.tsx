import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import '@ui5/webcomponents-icons/dist/icons/add.js';
import { Icon } from '@ui5/webcomponents-react/lib/Icon';
import React from 'react';

export default {
  title: 'UI5 Web Components / Icon',
  component: Icon
};

export const defaultStory = () => <Icon name={text('src', 'add')} onPress={action('onPress')} />;

defaultStory.story = {
  name: 'Default story'
};
