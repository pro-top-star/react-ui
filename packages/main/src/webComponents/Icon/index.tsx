import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import UI5Icon from '@ui5/webcomponents/dist/Icon';
import React, { FC } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface IconPropTypes extends WithWebComponentPropTypes {
  accessibleName?: string; //@generated
  name?: string; // @generated
  showTooltip?: boolean; // @generated
}

const Icon: FC<IconPropTypes> = withWebComponent<IconPropTypes>(UI5Icon);

Icon.defaultProps = {
  showTooltip: false
};

Icon.displayName = 'Icon';

export { Icon };
