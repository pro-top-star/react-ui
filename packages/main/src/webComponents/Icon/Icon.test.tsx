import { mountThemedComponent } from '@shared/tests/utils';
import { Icon } from '@ui5/webcomponents-react/lib/Icon';
import React from 'react';

describe('Icon', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<Icon />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
