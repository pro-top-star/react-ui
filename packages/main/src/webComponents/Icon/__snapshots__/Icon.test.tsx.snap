// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Basic Test (generated) 1`] = `<ui5-icon />`;
